Method `alloc` breaks moveit (compilation error)

**Describe the bug**
Method `alloc` breaks moveit (compilation error)

**To Reproduce**
```c++
struct DataBuf {                                                                                                                                      
    void alloc();                                                                                                                                     
};
```

```rust
generate!("DataBuf");
```

This cause the follow compilation error.
```
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0046]: not all trait items implemented, missing: `allocate_uninitialized_cpp_storage`
 --> /tmp/.tmpwE2POD/autocxx-ffi-default-gen.rs:1:1415
  |
1 | ...88849ba138 (this) }) } } } unsafe impl autocxx :: moveit :: MakeCppStorage for root :: DataBuf { unsafe fn free_uninitialized_cpp_stor...
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `allocate_uninitialized_cpp_storage` in implementation
  |
  = help: implement the missing item: `unsafe fn allocate_uninitialized_cpp_storage() -> *mut Self { todo!() }`
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

Reroducer: https://github.com/google/autocxx/pull/1319

**Expected behavior**

Compiles

**Additional context**

--- a/engine/src/conversion/analysis/allocators.rs
+++ b/engine/src/conversion/analysis/allocators.rs
@@ -101,11 +101,11 @@ fn create_alloc_and_free(ty_name: QualifiedName) -> impl Iterator<Item = Api<Pod
 }
 
 pub(crate) fn get_alloc_name(ty_name: &QualifiedName) -> QualifiedName {
-    get_name(ty_name, "alloc")
+    get_name(ty_name, "autocxx_alloc")
 }
 
 pub(crate) fn get_free_name(ty_name: &QualifiedName) -> QualifiedName {
-    get_name(ty_name, "free")
+    get_name(ty_name, "autocxx_free")
 }
 
 fn get_name(ty_name: &QualifiedName, label: &str) -> QualifiedName {

--- a/integration-tests/tests/integration_test.rs
+++ b/integration-tests/tests/integration_test.rs
@@ -12242,6 +12242,19 @@ fn test_ignore_va_list() {
 }
 
 #[test]
+fn test_badly_named_alloc() {
+    let hdr = indoc! {"
+        #include <stdarg.h>
+        class A {
+        public:
+            void alloc();
+        };
+    "};
+    let rs = quote! {};
+    run_test("", hdr, rs, &["A"], &[]);
+}
+
+#[test]
 fn test_cpp_union_pod() {
     let hdr = indoc! {"
         typedef unsigned long long UInt64_t;

